app/mixins/connection-type.js

```diff
@@ -0,0 +1,15 @@
+import Ember from 'ember';
+
+const {Mixin, inject, computed} = Ember;
+
+export default Mixin.create({
+	fastboot: inject.service(),
+	connection: computed('fastboot.isFastBoot', function () {
+		if (!this.get('fastboot.isFastBoot')) {
+			return navigator.connection || navigator.mozConnection || navigator.webkitConnection;
+		}
+	}),
+	effectiveConnectionType: computed('connection', function () {
```

Comment:
I don't think we need both computed properties. I think all the logic could be encapsulated in one computed property (effectiveConnectionType) since we do not need connection itself :)

Reply:
Don't you think is more readable this way?

Reply:
I agree with @bkoval . There is not need to add extra computed property.